sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.util;
+
+import org.apache.spark.sql.connector.expressions.Expression;
+import org.apache.spark.sql.connector.expressions.FieldReference;
+import org.apache.spark.sql.connector.expressions.GeneralScalarExpression;
+import org.apache.spark.sql.connector.expressions.LiteralValue;
+import org.apache.spark.sql.errors.QueryExecutionErrors;
+
+/**
+ * The builder to generate SQL from V2 expressions.
+ */
+public class V2ExpressionSQLBuilder {
+  public String build(Expression expr) throws Throwable {
+    if (expr instanceof LiteralValue) {
+      return visitLiteral((LiteralValue) expr);
+    } else if (expr instanceof FieldReference) {
+      return visitFieldReference((FieldReference) expr);
+    } else if (expr instanceof GeneralScalarExpression) {
+        GeneralScalarExpression e = (GeneralScalarExpression) expr;
+        String name = e.name();
+        switch (name) {
+          case "IS_NULL":
+            return visitIsNull(build(e.children()[0]));
+          case "IS_NOT_NULL":
+            return visitIsNotNull(build(e.children()[0]));
+          case "=":
+          case "!=":
+          case "<=>":
+          case "<":
+          case "<=":
+          case ">":
+          case ">=":
+          case "+":
+          case "-":
+          case "*":
+          case "/":
+          case "%":
+          case "&&":
+          case "||":
+          case "AND":
+          case "OR":
+          case "&":
+          case "|":
+          case "^":
+            return visitBinaryOperation(name, build(e.children()[0]), build(e.children()[1]));
+          case "NOT":
+            return visitNot(build(e.children()[0]));
+          case "CASE_WHEN":
+            return visitCaseWhen(e);
```

Comment:
shall we pass the children sql string? `visitCaseWhen(e.children.map(build)) // please use the proper jave 8 lambda syntax`